### ISSUE TITLE
Event when Menu closes

### ISSUE THREAD

**Issue by vital487** [SAP/ui5-webcomponents#5440]
Hi everyone,

I'm using the Menu component, but I need to execute a function when the Menu closes. I did not find any information about this event in the website. Can you please tell me if it exists or not and if it does not, do you know how to achieve the same result?

https://sap.github.io/ui5-webcomponents-react/?path=/docs/modals-popovers-menu--default-story

Thank you!

**Comment by nnaydenow** [comment]
Hi @vital487,

Why do you need the close / open state? Isn't it enough to know when the change happen? I'm confused what are you trying to achieve with this additional state.

Regards,
Nayden

**Comment by vital487** [comment]
Yes, if somehow I know when the change happens, it is enough, because then I can change my state.

Explaining more about how my app is working with the Menu: I have a context that, from other things, indicates if the Menu is visible or not. The Menu being visible or not changes how some components behave. That's why I need to have an event when the Menu closes, so that I can change the visible state of the Menu, so other components can change their behavior.

I'm not sure if this was confusing or not, but I can try to explain better if needed.

Thank you!

**Comment by nnaydenow** [comment]
Hi @SAP/ui5-webcomponents-topic-b,

Could you please check this issue?

Regards,
Nayden

**Comment by NHristov-sap** [comment]
Hi @vital487 ,

The ui5-menu component have  "item-click" event which is fired when an item is clicked (selected). The event returns as detail the text of the selected item and the selected item itself.
Is this enough for your purposes?

Best Regards,
Nikolay Hristov

**Comment by vital487** [comment]
Hi @NHristov-sap,

It is not enough because there are other ways to close the Menu, like the Escape button and clicking outside the Menu. I created some code to overcome this issue. It is pasted below:

`
useEffect(() => {
		if (contextMenu.isVisible && ref.current) {
			ref.current.showAt(contextMenu.event.target);
		}

		const handleClickOutside = (e) => {
			if (ref.current && e.target && !ref.current.contains(e.target) && e.target.tagName !== 'UI5-STATIC-AREA-ITEM' && contextMenu.isVisible) {
				console.log(e.target.tagName, ref.current, e.target)
				contextMenu.close();
			}
		}

		const handleKeyboard = (e) => {
			if (e.key === 'Escape' && contextMenu.isVisible) {
				contextMenu.close();
			}
		}

		document.addEventListener('mousedown', handleClickOutside);
		document.addEventListener('keydown', handleKeyboard);

		return () => {
			document.removeEventListener('mousedown', handleClickOutside);
			document.removeEventListener('keydown', handleKeyboard);
		};
}, [contextMenu.isVisible, contextMenu.event]);
`

Basically this useEffect runs when the Menu needs to open. I handle the click outside the Menu and the keyboard (Escape).

Although this code works, this condition looks bad `e.target.tagName !== 'UI5-STATIC-AREA-ITEM'`. The problem is that it was not working with just this condition `!ref.current.contains(e.target)`, so I needed to add that one.

The ref points to the Menu.

Thank you!

**Comment by vital487** [comment]
Hi @NHristov-sap,

Is this issue going to be solved someday or since I posted a fix it will be ignored?

Thank you!

**Comment by NHristov-sap** [comment]
Hi @vital487 ,

Currently your request (and some other ui5-menu related) is under development and we hope all will be available very soon.

Best Regards,
Nikolay Hristov
